draft-dekater-scion-dataplane.md

```diff
@@ -186,6 +186,8 @@ SCION leverages source-based path selection, where path information is embedded
 
 **Info Field (INF)**: Each path-segment construction beacon (PCB) contains a single info field, which provides basic information about the PCB. Together with hop fields (HFs), info fields are used to create forwarding paths.
 
+**Interface Identifier (Interface ID)**: 16 bit identifier that designates a SCION interface at the end of a link connecting two SCION ASes. Each interface belongs to one border router. Hop fields describe the traversal of an AS by a pair of interface IDs (the ingress and egress interfaces). The Interface ID MUST be unique within each AS, with the exception of ID 0 that denotes a connection between a border router and the AS's internal network. Each border router manages an instance of interface 0.
```

Comment:
We actually describe Interface ID 0 in the _One-Hop Path Type_  section. 
Maybe we could add a reference?   

Also, is interface ID 0 used for the last hop? In the example in _Step-by-Step Explanation_
```
      R4 will then also realize, based on the fields
      CurrHF and SegLen in the SCION header, that the packet has reached
      the last hop in its SCION path.  Therefore, instead of stepping up
      the pointers to the next info or hop field, router R4 inspects the
      SCION destination address and extracts the endpoint address
      192.0.2.7. 
 ```

Reply:
> We actually describe Interface ID 0 in the One-Hop Path Type section.

The One hop path is not very special here; we also use 0 interface in the first/last hop field of normal SCION paths.

> Also, is interface ID 0 used for the last hop?

During path segment construction, the 0 interface is used in the first/last hop field. BUT in the path-segment-combined dataplane path, the 0 does not always/only occur on the last hop.
- Hops with 0 interface occurs "in the middle", at segment cross overs.
- Typically 0 will occur at the end of a down segment, but it's legal to cut off the end of a segment.
- Paths to a parent AS below the core consist of only an up-segment that is cut short. These don't end with 0.

Reply:
Thanks for the explanation! @jiceatscion can you remove mentions of this use of interface ID 0 from the draft?